packages/expo-cli/src/commands/utils/validateDependenciesVersions.ts

```diff
@@ -92,7 +92,18 @@ async function resolvePackageVersionsAsync(
 }
 
 async function getPackageVersionAsync(projectRoot: string, packageName: string): Promise<string> {
-  const packageJsonPath = resolveFrom.silent(projectRoot, `${packageName}/package.json`);
+  let packageJsonPath: string | undefined;
+  try {
+    packageJsonPath = resolveFrom(projectRoot, `${packageName}/package.json`);
+  } catch (error: any) {
+    // This is a workaround for packages using `exports`. If this doesn't
+    // include `package.json`, we have to use the error message to get the location.
+    if (error.code === 'ERR_PACKAGE_PATH_NOT_EXPORTED') {
+      packageJsonPath = error.message.match(/ in (.*)$/i)?.[1];
+    } else {
```

Comment:
is this else block necessary? if `resolveFrom` throws an error then i believe `resolveFrom.silent` will return `undefined` for the same params. so we'd just be setting `packageJsonPath` to `undefined` here. i'm probably not thinking of some other case that this is meant to handle though 😅

Reply:
I somehow thought that `resolve-from` also handled another error, but that seems to be part of the lookup logic. You are right, should be able to just put null or undefined there.